src/components/widgets/diagnostics/config/MetricsConfigStep.vue

```diff
@@ -99,10 +99,10 @@
 
               <app-setting :title="$t('app.setting.label.fill_color')">
                 <app-color-picker
-                  :primary="metric.style.fillColor ?? metric.style.lineColor"
+                  :value="metric.style.fillColor ?? metric.style.lineColor"
                   :title="metric.name"
                   dot
-                  @change="handleColorChange('fillColor', metric, $event)"
+                  @input="metric.style.fillColor = $event"
```

Comment:
The fill color selection doesn't update when the color is changed - triggering an update for the whole style prop seems to fix this:
```suggestion
                  @input="metric.style = { ...metric.style, fillColor: $event }"
```

Reply:
From my tests, I don't think this is necessary - this is why I changed the `fillColor` default from `undefined` to `null` as the previous is not reactive and the latter is!

Reply:
That fixes it for new metrics, but the old ones have `undefined` stored in the moonraker DB ;)

Reply:
well, no, of course they don't, as undefined isn't a JSON type.. i'll take another look at it :D

Reply:
Ah yeah, the `fillColor` key is just missing in the stored data. Not sure if it makes sense to write an extra data migration step for this though, I think just updating the whole `style` prop is fine here

Reply:
I've added a migration step to make sure that any existing `fillColor = undefined` gets set to `null` instead, so that should solve the problem for existing data!